src/runtime/crt/contrib/stm32/ai_runtime_api.h

```diff
@@ -0,0 +1,226 @@
+AI_STATIC AI_INLINE uint32_t get_tensor_size(const ai_tensor* tensor) {
+  const DLTensor* t = &tensor->dltensor;
+  uint32_t size = 1;
+  for (int i = 0; i < t->ndim; ++i) {
+    size *= t->shape[i];
+  }
+  size *= (t->dtype.bits * t->dtype.lanes + 7) / 8;
+  return size;
+}
+
+// =======================================================
+//                    ai_network_info
+// =======================================================
+
+typedef struct {
+  const char* name;
+  const char* datetime;
+  const char* revision;
+  const char* tool_version;
+  const char* api_version;
+  uint16_t n_nodes;
+  uint8_t n_inputs;
+  uint8_t n_outputs;
+  uint32_t activations_size;
+  uint32_t params_size;
+  ai_ptr activations;
+  ai_tensor** inputs;
+  ai_tensor** outputs;
+  const ai_ptr (*ai_get_params)(void);
+  ai_status (*ai_create)(const ai_ptr weights, const ai_ptr activations);
+  ai_status (*ai_destroy)();
+  ai_status (*ai_run)(ai_tensor* input[], ai_tensor* output[]);
+} ai_model_info;
+
+#define AI_MODEL_name(x) (x->name)
+#define AI_MODEL_datetime(x) (x->datetime)
+#define AI_MODEL_revision(x) (x->revision)
+#define AI_MODEL_tool_version(x) (x->tool_version)
+#define AI_MODEL_api_version(x) (x->api_version)
+#define AI_MODEL_n_nodes(x) (x->n_nodes)
+#define AI_MODEL_n_inputs(x) (x->n_inputs)
+#define AI_MODEL_n_outputs(x) (x->n_outputs)
+#define AI_MODEL_activations_size(x) (x->activations_size)
+#define AI_MODEL_params_size(x) (x->params_size)
+#define AI_MODEL_inputs(x) (x->inputs)
+#define AI_MODEL_outputs(x) (x->outputs)
+#define AI_MODEL_activations(x) (x->activations)
+
+// =======================================================
+//                         Iterator
+//
+//   Usage:
+//
+//     for (ai_models_iterator it = ai_models_iterator_begin();
+//          it != ai_models_iterator_end();
+//          it = ai_models_iterator_next(it)) {
+//       const char * name = ai_models_iterator_value(it);
+//     }
+//
+// =======================================================
+
+typedef uint32_t ai_model_iterator;
+
+ai_model_iterator ai_model_iterator_begin();
+ai_model_iterator ai_model_iterator_next(ai_model_iterator it);
+ai_model_iterator ai_model_iterator_end();
+ai_model_info* ai_model_iterator_value(ai_model_iterator it);
+
+// =======================================================
+//                    External Interface
+// =======================================================
+
+ai_status ai_create(ai_model_info* nn, void* context, ai_handle* handle);
+
+ai_status ai_destroy(ai_handle handle);
+
+const char* ai_get_error(ai_handle handle);
+
+int32_t ai_get_input_size(ai_handle handle);
+
+int32_t ai_get_output_size(ai_handle handle);
+
+ai_tensor* ai_get_input(ai_handle handle, int32_t index);
+
+ai_tensor* ai_get_output(ai_handle handle, int32_t index);
+
+ai_status ai_run(ai_handle handle);
+
+//
+// Additional methods
+//
+const char* ai_get_name(ai_handle handle);
```

Comment:
could you ensure these are tested? in particular i only see usages of name, get_activations, and get_params in the tests